tools/packaging/packaging.psm1

```diff
@@ -246,255 +260,301 @@ function New-UnixPackage {
         [Parameter(Mandatory)]
         [string]$Version,
 
+        # Package iteration version (rarely changed)
+        # This is a string because strings are appended to it
+        [string]$Iteration = "1",
+
         [Switch]
         $Force
     )
 
-    # Validate platform
-    $ErrorMessage = "Must be on {0} to build '$Type' packages!"
-    switch ($Type) {
-        "deb" {
-            $verboseMsg = "Building for Ubuntu {0}.04!"
-            if (!$Environment.IsUbuntu) {
-                throw ($ErrorMessage -f "Ubuntu")
-            } elseif ($Environment.IsUbuntu14) {
-                Write-Verbose ($verboseMsg -f "14")
-            } elseif ($Environment.IsUbuntu16) {
-                Write-Verbose ($verboseMsg -f "16")
-            }
-        }
-        "rpm" {
-            if (!$Environment.IsRedHatFamily) {
-                throw ($ErrorMessage -f "Redhat Family")
-            }
-        }
-        "osxpkg" {
-            if (!$Environment.IsMacOS) {
-                throw ($ErrorMessage -f "macOS")
-            }
+    DynamicParam {
+        if ($Type -eq "deb") {
+            # Add a dynamic parameter '-Distribution' when the specified package type is 'deb'.
+            # The '-Distribution' parameter can be used to indicate which Debian distro this pacakge is targeting.
+            $ParameterAttr = New-Object "System.Management.Automation.ParameterAttribute"
```

Comment:
use ```[System.Management.Automation.ParameterAttribute]::new()``` instead

Reply:
We use `New-Object` in build.psm1, and thus I used `New-Object` here to keep consistent.